python/lsst/cell_coadds/_cellCoaddBuilder.py

```diff
@@ -0,0 +1,385 @@
+        name="skyMap",
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    cellCoadd = cT.Output(
+        doc="Coadded image",
+        name="{outputCoaddName}CellCoaddPickled",
+        storageClass="MultipleCellCoadd",
+        dimensions=("tract", "patch", "skymap", "band", "instrument"),
+    )
+
+
+class MultipleCellCoaddBuilderConfig(
+    pipeBase.PipelineTaskConfig, pipelineConnections=MultipleCellCoaddBuilderConnections
+):
+    """Configuration parameters for the `MultipleCellCoaddBuilderTask`."""
+
+    inputType = pexConfig.ChoiceField(
+        doc="Type of input dataset",
+        dtype=str,
+        allowed={"inputWarps": "Use warps", "calexps": "Use calexps"},
+        default="calexps",
+    )
+
+    psf_dimensions: pexConfig.Field = pexConfig.Field(
+        doc="Dimensions of the PSF image",
+        dtype=int,
+        default=41,
+    )
+
+    singleCellCoaddBuilder = singleCellCoaddBuilderRegistry.makeField(
+        doc="Coaddition algorithm to use. See `SingleCellCoaddBuilderTask` for details",
+        optional=False,
+    )
+
+
+class MultipleCellCoaddBuilderTask(pipeBase.PipelineTask):
+    """Task to build cell-based coadded images.
+
+    This is the pipeline task that needs to be called from the pipeline. It
+    contains ``singleCellCoaddBuilder`` as a subtask which must implement a
+    concrete coaddition algorithm in its ``run`` method. The images to be
+    coadded can either be of type ``calexp`` or ``warp``, as specified in the
+    ``inputType`` configuration field. A ``skymap`` with ``cells`` tract
+    builder must be passed. For each cell to be coadded, the task will query
+    the butler for all the input images that completely overlap the cell's
+    outer bounding box and passed them to the ``run`` method of the
+    ``singleCellCoaddBuilder``.
+
+    See also
+    --------
+    SingleCellCoaddBuilderTask
+    """
+
+    ConfigClass = MultipleCellCoaddBuilderConfig
+    _DefaultName = "multipleCellCoaddBuilder"
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.makeSubtask(name="singleCellCoaddBuilder")
+        self.singleCellCoaddBuilder: SingleCellCoaddBuilderTask
+
+    def runQuantum(  # type: ignore[override]
+        self,
+        butlerQC: pipeBase.ButlerQuantumContext,
+        inputRefs: pipeBase.InputQuantizedConnection,
+        outputRefs: pipeBase.OutputQuantizedConnection,
+    ) -> MultipleCellCoadd:
+        # Docstring inherited.
+        self.config: MultipleCellCoaddBuilderConfig
+        inputs: dict = butlerQC.get(inputRefs)
+        skyMap = inputs.pop("skyMap")  # skyInfo below will contain this skyMap
+        # Ideally, we should do this check early on.
+        # But skyMap is not available during config time.
+        if not skyMap.config.tractBuilder.name == "cells":
+            raise TypeError("skyMap is not a cells skyMap")
```

Comment:
[`lsst.pipe.base.InvalidQuantumError`](https://github.com/lsst/pipe_base/blob/454b09c8d2a09001faadd11c30e2a407381911b8/python/lsst/pipe/base/_status.py#L71-L92) would be a good choice here